src/platform/APM32/pwm_output_apm32.c

```diff
@@ -151,46 +151,48 @@ static motorVTable_t motorPwmVTable = {
     .shutdown = pwmShutdownPulsesForAllMotors,
     .convertExternalToMotor = pwmConvertFromExternal,
     .convertMotorToExternal = pwmConvertToExternal,
+    .requestTelemetry = NULL,
+    .isMotorIdle = NULL,
 };
 
-motorDevice_t *motorPwmDevInit(const motorDevConfig_t *motorConfig, uint16_t idlePulse, uint8_t motorCount, bool useUnsyncedPwm)
+motorDevice_t *motorPwmDevInit(const motorDevConfig_t *motorConfig, uint16_t idlePulse, uint8_t motorCount, bool useUnsyncedUpdate)
 {
     memset(motors, 0, sizeof(motors));
 
     motorPwmDevice.vTable = motorPwmVTable;
 
     float sMin = 0;
     float sLen = 0;
-    switch (motorConfig->motorPwmProtocol) {
+    switch (motorConfig->motorProtocol) {
     default:
-    case PWM_TYPE_ONESHOT125:
+    case MOTOR_PROTOCOL_ONESHOT125:
         sMin = 125e-6f;
         sLen = 125e-6f;
         break;
-    case PWM_TYPE_ONESHOT42:
+    case MOTOR_PROTOCOL_ONESHOT42:
         sMin = 42e-6f;
         sLen = 42e-6f;
         break;
-    case PWM_TYPE_MULTISHOT:
+    case MOTOR_PROTOCOL_MULTISHOT:
         sMin = 5e-6f;
         sLen = 20e-6f;
         break;
-    case PWM_TYPE_BRUSHED:
+    case MOTOR_PROTOCOL_BRUSHED:
         sMin = 0;
-        useUnsyncedPwm = true;
+        useUnsyncedUpdate = true;
         idlePulse = 0;
         break;
-    case PWM_TYPE_STANDARD:
+    case MOTOR_PROTOCOL_PWM50HZ :
         sMin = 1e-3f;
         sLen = 1e-3f;
-        useUnsyncedPwm = true;
+        useUnsyncedUpdate = true;
         idlePulse = 0;
         break;
     }
 
     motorPwmDevice.vTable.write = pwmWriteStandard;
     motorPwmDevice.vTable.decodeTelemetry = motorDecodeTelemetryNull;
-    motorPwmDevice.vTable.updateComplete = useUnsyncedPwm ? motorUpdateCompleteNull : pwmCompleteOneshotMotorUpdate;
+    motorPwmDevice.vTable.updateComplete = useUnsyncedUpdate ? motorUpdateCompleteNull : pwmCompleteOneshotMotorUpdate;
```

Comment:
(this will prevent using const vTable)

Reply:
I have moved it all to one instance of motorDevice_t and vTable should now be const.